#3163 Don't include the DwcOccurrence id column in DwC exports

kleintom opened the pull request.

mjy commented:
@kleintom This is by design.  The id is not persistent.  To recover records we need to reference the occurrenceID.

mjy commented:
@kleintom )(@*$

mjy commented:
@kleintom This may be introducing failures: https://github.com/SpeciesFileGroup/taxonworks/actions/runs/7587358876/job/20667528644?pr=3789#step:21:74

kleintom commented:
That worked out great actually, it
* caught a bug I had introduced: `DwcOccurrences.meta_fields` was opening the newly created csv file, yanking out the headers into an array, and then shifting off the first one, which used to be `id`. Now the shift is unnecessary. That fixed all but one of the failing tests (those were all checking meta_fields).
* The other fixed failing test was checking the header fields in an exported dwc csv file - the check now correctly checks that `id` is not included.
(I squashed the commits so there wouldn't be failing tests on any of them.)
